josh-core/src/history.rs

```diff
@@ -75,32 +75,44 @@ pub fn walk2(
 
 fn find_unapply_base(
     transaction: &cache::Transaction,
-    bm: &mut HashMap<git2::Oid, git2::Oid>,
+    // Used as a cache to avoid re-applying the filter to the same commit -
+    // this function is called during revwalk so there be a lot of repeated
+    // calls
+    filtered_to_original: &mut HashMap<git2::Oid, git2::Oid>,
     filter: filter::Filter,
+    // "contained in" refers to OID of the commit in the original (unfiltered) history
```

Comment:
That's not what `contained_in` means. What it does mean: When building the `bm`/`filtered_to_original` mapping use this as a starting point for the search for orginals. Meaning if there are multiple originals that map to the same filtered commit (which is common) use one that is reachable from `contained_in`. Or in other words one that is contained in the the history of `contained_in`.